### PR TITLE
New shell redirect

### PR DESCRIPTION
**Pull request by shrutiyer**

### CODE REVIEW

File: snl_shell/snl.c [hunk 1]
```diff
@@ -147,6 +151,18 @@ int snl_detect_pipe(char** args){
   return 0; // no pipe
 }
 
+int snl_detect_redirect(char** args) {
+  for (int i = 0; i < MAXARGS; i++) { //TODO: make this not maxargs
```
Comment (LucyWilcox) [inline snl_shell/snl.c:155]: Wait, why should this not be maxargs?

Comment (LucyWilcox) [reply snl_shell/snl.c:155]: Oh is this just saying it would be better if we dynamically read our args?

**Comment by LucyWilcox** [comment]
Looks good to me.

**Comment by shrutiyer** [comment]
@nmohamed merge it if it looks good to you

File: snl_shell/snl.c [hunk 2]
```diff
@@ -147,6 +151,18 @@ int snl_detect_pipe(char** args){
...
+    if(args[i] == NULL){
+      return 0; // reached end, no pipe
+    }
+    if (strcmp(args[i],">") == 0){
+      return i; // yes pipe
+    }
+  }
+  return 0; // no pipe
+}
+
```
Comment (nmohamed) [inline snl_shell/snl.c:165]: i would add some error checking to make sure the user doesn't input both | and </>

File: snl_shell/snl.c [hunk 3]
```diff
@@ -168,6 +184,11 @@ int snl_execute(char** args) {
   if (snl_detect_pipe(args) == 1){
     return snl_forkpipe(args);
   }
+  if (snl_detect_redirect(args) != 0) {
+    // REDIRECT
+    printf("REDIRECT\n");
```
Comment (nmohamed) [inline snl_shell/snl.c:189]: remove the printf and either remove the comment or add one for pipe as well

File: snl_shell/snl.c [hunk 4]
```diff
@@ -339,6 +360,43 @@ int snl_forkpipe(char* args[]){
   return 1;
 }
 
+void get_command(char* args[], char* command[], int redirect_index) {
+  for (int i=0; i<redirect_index; i++) {
+    command[i] = args[i];
+    printf("ARGS IN COMMAND %s\n", command[i]);
```
Comment (nmohamed) [inline snl_shell/snl.c:366]: remove printfs

File: snl_shell/snl.c [hunk 5]
```diff
@@ -339,6 +360,43 @@ int snl_forkpipe(char* args[]){
...
+int snl_fork_redirect(char* args[], int redirect_index) {
+  int file_descriptor;
+  char* filename = args[redirect_index+1];
+  char* command[100];
+
+  get_command(args, command, redirect_index);
+
+  pid_t pid = fork();
+  if (pid == -1) {
+    printf("Fork could not create a child process\n");
```
Comment (nmohamed) [inline snl_shell/snl.c:380]: printf

File: snl_shell/snl.c [hunk 6]
```diff
@@ -339,6 +360,43 @@ int snl_forkpipe(char* args[]){
...
+  } else if (pid == 0) {
+
+    // Open a  file to write to it
+    file_descriptor = open(filename, O_CREAT | O_TRUNC | O_WRONLY, 0600);
+    if (file_descriptor < 0) {
+      perror("Incorrect file given for redirect");
```
Comment (nmohamed) [inline snl_shell/snl.c:386]: can you make this "snl: Incorrect file given for redirect" for consistency?

File: snl_shell/snl.c [hunk 7]
```diff
@@ -339,6 +360,43 @@ int snl_forkpipe(char* args[]){
...
+    }
+		// Replace standard output with appropriate file
+		dup2(file_descriptor, 1);
+    close(file_descriptor);
+
+    if (execvp(command[0], command) == -1){
+      perror("snl fork redirect");
+  		kill(getpid(),SIGTERM);
```
Comment (nmohamed) [inline snl_shell/snl.c:394]: there's some wierd tab spacing going on

**Review by nmohamed — APPROVED** [review]
just some formatting stuff really, but otherwise it's good. the error checking thing could be a separate pr if you just want to merge this now

**Pull request merged**